src/policy.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2014 The Bitcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+// NOTE: This file is intended to be customised by the end user, and includes only local node policy logic
+
+#include "policy.h"
+
+#include "amount.h"
+#include "tinyformat.h"
+#include "ui_interface.h"
+#include "utilstrencodings.h"
+
+bool fIsBareMultisigStd = true;
+/** Fees smaller than this (in satoshi) are considered zero fee (for relaying and mining) */
+CFeeRate minRelayTxFee = CFeeRate(1000);
+static const unsigned int MAX_OP_RETURN_RELAY = 40; //! bytes
+
+/** Declaration of Standard Policy implementing CPolicy */
+class CStandardPolicy : public CPolicy
+{
+protected:
+    unsigned nMaxDatacarrierBytes;
+public:
+    CStandardPolicy() : nMaxDatacarrierBytes(MAX_OP_RETURN_RELAY) {};
+
+    virtual void InitFromArgs(const std::map<std::string, std::string>&);
+    virtual bool ValidateScript(const CScript&, txnouttype&) const;
+};
+
+/** Global variables and their interfaces */
+
+static CStandardPolicy standardPolicy;
+
+static CPolicy* pCurrentPolicy = 0;
+
+CPolicy& Policy(std::string policy)
+{
+    if (policy == "standard")
+        return standardPolicy;
+    throw std::runtime_error(strprintf(_("Unknown policy '%s'"), policy));
+}
+
+void SelectPolicy(std::string policy)
+{
+    pCurrentPolicy = &Policy(policy);
+}
+
+const CPolicy& Policy()
+{
+    assert(pCurrentPolicy);
+    return *pCurrentPolicy;
+}
+
+std::string GetPolicyUsageStr()
+{
+    std::string strUsage = "";
+    strUsage += "  -datacarrier           " + strprintf(_("Relay and mine data carrier transactions (default: %u)"), 1) + "\n";
+    strUsage += "  -datacarriersize       " + strprintf(_("Maximum size of data in data carrier transactions we relay and mine (default: %u)"), MAX_OP_RETURN_RELAY) + "\n";
+    strUsage += "  -policy                " + strprintf(_("Select a specific type of policy (default: %s)"), "standard") + "\n";
+    return strUsage;
+}
+
+void InitPolicyFromArgs(const std::map<std::string, std::string>& mapArgs)
+{
+    SelectPolicy(GetArg("-policy", "standard", mapArgs));
+    pCurrentPolicy->InitFromArgs(mapArgs);
+}
+
+/** CStandardPolicy implementation */
+
+void CStandardPolicy::InitFromArgs(const std::map<std::string, std::string>& mapArgs)
+{
+    if (GetArg("-datacarrier", true, mapArgs))
+        nMaxDatacarrierBytes = GetArg("-datacarriersize", nMaxDatacarrierBytes, mapArgs);
+    else
+        nMaxDatacarrierBytes = 0;
+}
+
+bool CStandardPolicy::ValidateScript(const CScript& scriptPubKey, txnouttype& whichType) const
```

Comment:
The fact that we're using Accept in main may actually be a reason against also using it in policy rather than a reason in favor.
If you look at #5768 you may understand why I definitely prefer ValidateX over AcceptX at this point. I'm not changing it unless you get more people to support this bikeshed.